hushline/settings.py

```diff
@@ -224,57 +224,55 @@ def index() -> str | Response:
         directory_visibility_form = DirectoryVisibilityForm()
         profile_form = ProfileForm()
 
-        # Check if the bio update form was submitted
-        if (
-            request.method == "POST"
-            and "update_bio" in request.form
-            and profile_form.validate_on_submit()
-        ):
-            user.bio = request.form["bio"]
-            user.extra_field_label1 = profile_form.extra_field_label1.data.strip()
-            user.extra_field_value1 = profile_form.extra_field_value1.data.strip()
-            user.extra_field_label2 = profile_form.extra_field_label2.data.strip()
-            user.extra_field_value2 = profile_form.extra_field_value2.data.strip()
-            user.extra_field_label3 = profile_form.extra_field_label3.data.strip()
-            user.extra_field_value3 = profile_form.extra_field_value3.data.strip()
-            user.extra_field_label4 = profile_form.extra_field_label4.data.strip()
-            user.extra_field_value4 = profile_form.extra_field_value4.data.strip()
-            db.session.commit()
-            flash("👍 Bio updated successfully.")
-            return redirect(url_for("settings.index"))
+        # Handle form submissions
+        if request.method == "POST":
+            # Update bio and custom fields
+            if "update_bio" in request.form and profile_form.validate_on_submit():
+                user.bio = request.form["bio"].strip()
+                user.extra_field_label1 = profile_form.extra_field_label1.data.strip()
+                user.extra_field_value1 = profile_form.extra_field_value1.data.strip()
+                user.extra_field_label2 = profile_form.extra_field_label2.data.strip()
+                user.extra_field_value2 = profile_form.extra_field_value2.data.strip()
+                user.extra_field_label3 = profile_form.extra_field_label3.data.strip()
+                user.extra_field_value3 = profile_form.extra_field_value3.data.strip()
+                user.extra_field_label4 = profile_form.extra_field_label4.data.strip()
+                user.extra_field_value4 = profile_form.extra_field_value4.data.strip()
+
+                # Trigger the rel=me verification for each URL field
+                for i in range(1, 5):
+                    url_to_verify = getattr(user, f"extra_field_value{i}", "").strip()
+                    if url_to_verify:
+                        try:
+                            response = requests.get(url_to_verify, timeout=5)
+                            response.raise_for_status()
+                            profile_url = f"https://tips.hushline.app/to/{user.primary_username}"
+                            if re.search(
```

Comment:
Instead of searching for the rel="me" link using regex, it would be better to use an HTML parsing library like BeautifulSoup (the python module is bs4, and we actually already use bs4 as a dev dependency because I used it in tests, but we could move it to a main dependency).

The reason is because there's a lot of different ways that the link could be valid HTML but the regex wouldn't notice it. For example, these are all valid:

```html
<a href="https://tips.hushline.app/to/admin" rel="nofollow me">
<a href='https://tips.hushline.app/to/admin' rel='me'>
<a href="https://tips.hushline.app/to/admin" rel=me>
<a
    href = "https://tips.hushline.app/to/admin"
    rel = "me"
>
<  a       href   =      "https://tips.hushline.app/to/admin" rel                      =                                         "me"    >
```

(If you add a link to your github profile, it adds `rel="nofollow me"`.)

Here's how you can use bs4:

```py
import requests
import bs4

def verify(field_value_url, profile_url):
    r = requests.get(field_value_url)
    soup = bs4.BeautifulSoup(r.text, "html.parser")
    for link in soup.find_all("a"):
        if link.get("href") == profile_url and "me" in link.get("rel"):
            print("Found rel=me link to profile URL")

verify("https://glennsorrentino.com/", "https://tips.hushline.app/to/admin")
verify("https://github.com/micahflee", "https://infosec.exchange/@micahflee")
```

Reply:
Changes in bcae3d3